File: python/novax_sdk/pay/receipt/receipt_protocols_request.py
from typing import Any, Optional

from novax_sdk.pay.model import PayProtocolsResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


class ReceiptProtocolsRequest(ApiRequest):
    """GET /pay/v3/protocols — type: 1=dynamic address, 6=fixed address."""

    def __init__(self, type: int = 1) -> None:
        self._type = type

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/v3/protocols"

    def query_params(self) -> dict[str, Any]:
        return {"type": self._type}

    @property
    def response_type(self):
        return list[PayProtocolsResp]

File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/receipt/ReceiptProtocolsRequest.java
package com.novax.sdk.pay.receipt;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.PayProtocolsResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/**
 * GET /pay/v3/protocols — list chains supported for receipt orders.
 * {@code type}: 1 = dynamic address, 6 = fixed address. Defaults to 1.
 */
@Getter
@Builder
public class ReceiptProtocolsRequest extends AbstractApiRequest<List<PayProtocolsResp>> {

    private final Integer type;

    @Override
    public HttpMethod method() {
        return HttpMethod.GET;
    }

    @Override
    public String path() {
        return "/pay/v3/protocols";
    }

    @Override
    public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        q.put("type", type != null ? type : 1);
        return q;
    }

    @Override
    public TypeRef<List<PayProtocolsResp>> responseType() {
        return new TypeRef<>() {
        };
    }
}
